src/main/java/com/dataloom/linking/requests/LinkingRequest.java

```diff
@@ -0,0 +1,32 @@
+package com.dataloom.linking.requests;
+
+import java.util.Set;
+import java.util.UUID;
+
+import com.dataloom.client.serialization.SerializationConstants;
+import com.dataloom.edm.set.LinkingEntitySet;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class LinkingRequest {
+    private final LinkingEntitySet linkingEntitySet;
+    private final Set<UUID>        resultPropertyTypeIds;
+
+    @JsonCreator
+    public LinkingRequest(
+            @JsonProperty( SerializationConstants.LINKING_ENTITY_SET_FIELD ) LinkingEntitySet linkingEntitySet,
+            @JsonProperty( SerializationConstants.PROPERTY_TYPES ) Set<UUID> resultPropertyTypeIds ) {
+        this.linkingEntitySet = linkingEntitySet;
+        this.resultPropertyTypeIds = resultPropertyTypeIds;
```

Comment:
Can `resultPropertyTypeIds` be an empty set?

Reply:
theoretically, but that wouldn't make very much sense. it would just be an empty entity set